src/app/services/courses-store.service.ts

```diff
@@ -1,44 +1,98 @@
 import { Injectable } from '@angular/core';
+import { BehaviorSubject, finalize, Observable, tap } from 'rxjs';
+import { Course } from '@shared/models/course.model';
+import { CoursesService } from '@app/services/courses.service';
+import { Author } from '@shared/models/author.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CoursesStoreService {
+  private isLoading$$ = new BehaviorSubject<boolean>(false);
+  private selectedCourse$$ = new BehaviorSubject<Course | null>(null);
+  private selectedAuthor$$ = new BehaviorSubject<Author | null>(null);
+  private courses$$ = new BehaviorSubject<Course[]>([]);
+  private authors$$ = new BehaviorSubject<Author[]>([]);
+
+  public isLoading$ = this.isLoading$$.asObservable();
+  public selectedCourse$ = this.selectedCourse$$.asObservable();
+  public selectedAuthor$ = this.selectedAuthor$$.asObservable();
+  public courses$ = this.courses$$.asObservable();
+  public authors$ = this.authors$$.asObservable();
+
+  constructor(private coursesService: CoursesService) {}
+
   getAll() {
     // Add your code here
+    this.isLoading$$.next(true);
+
+    this.coursesService
+      .getAll()
+      .pipe(finalize(() => this.isLoading$$.next(false)))
+      .subscribe(courses => {
+        this.courses$$.next(courses.result);
+      });
   }
 
-  createCourse(course: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  createCourse(course: Course) {
+    this.coursesService.createCourse(course).subscribe(course => {
+      this.courses$$.next([...this.courses$$.getValue(), course]);
```

Comment:
Well, theoretically this is fine, in a scalable application this might not be the case. Suppose that multiple courses are added by multiple users simultaneously. You are adding the new one to the existing list, which will not include the new courses created by others. Try to get the full courses list after a new one is created. Or you might even want to do that but get the full list once navigating back to courses page.

Reply:
Well, i used the getAll method after the course is created, i don't know if it should do the job, hope so.